kernel/mmu.c

```diff
@@ -781,10 +781,13 @@ void z_phys_map(uint8_t **virt_ptr, uintptr_t phys, size_t size, uint32_t flags)
 		 * Basically if either end of region is within
 		 * virtual memory space, we need to mark the bits.
 		 */
-		if (((dest_addr >= Z_VIRT_RAM_START) &&
-		     (dest_addr < Z_VIRT_RAM_END)) ||
-		    (((dest_addr + aligned_size) >= Z_VIRT_RAM_START) &&
-		     ((dest_addr + aligned_size) < Z_VIRT_RAM_END))) {
+
+		if (IN_RANGE(aligned_phys,
+			      (uintptr_t)Z_VIRT_RAM_START,
+			      (uintptr_t)(Z_VIRT_RAM_END - 1)) ||
+		    IN_RANGE(aligned_phys + aligned_size - 1,
```

Comment:
I think you can skip the minus 1 here, as you have remove 1 byte from the end value. Or else the comparison would be off by 1 byte.

Reply:
I think that it is needed. The _IN_RANGE()_ macro uses a _<=_ comparison for the upper bound and _Z_VIRT_RAM_END_ is simply "start address + size". Furthermore, in the corner case where that would take us to the very end of the addresses, without the "-1", our end address would wrap around to zero. With the "-1", we ensure that we remain within the boundaries.

Reply:
Ahh... pointer math is confusing.